### PR TITLE
Binary hp loader

### PR DESCRIPTION
**Pull request by RaphaelKreft**
EG's branch to implement LineHeatmap Dataloading

### CODE REVIEW

File: .vscode/launch.json
Comment (RaphaelKreft) [inline .vscode/launch.json:1]: Exclude this folder from git by adding an entry in .gitignore for it

File: =
Comment (RaphaelKreft) [inline =:1]: delete this file

File: gluefactory/settings.py
Comment (RaphaelKreft) [inline gluefactory/settings.py:1]: please don't change settings file in this PR

Comment (RaphaelKreft) [reply gluefactory/settings.py:1]: we don't want to change this as everybody has its own settings adapted to their machine so we better keep this as is

File: gluefactory/datasets/oxford_paris_mini_1view_jpldd.py [hunk 1]
```diff
@@ -337,6 +406,12 @@ def __getitem__(self, idx):
         full_artificial_img_path = self.img_dir / self.image_sub_paths[idx]
         folder_path = full_artificial_img_path.parent / full_artificial_img_path.stem
         img = self._read_image(folder_path)
+        print(f"Image shape: {img.shape} and type: {img.dtype}")
```
Comment (RaphaelKreft) [inline gluefactory/datasets/oxford_paris_mini_1view_jpldd.py]: remove print

File: gluefactory/datasets/oxford_paris_mini_1view_jpldd.py [hunk 2]
```diff
@@ -337,6 +406,12 @@ def __getitem__(self, idx):
...
+        # apply augmentation in try/ catch block
+        try:
+            img = self.augmentation(image=img, return_tensor=True).permute(1,2,0)
+        except Exception as e:
+            print(f"Error in augmentation: {e}")
```
Comment (RaphaelKreft) [inline gluefactory/datasets/oxford_paris_mini_1view_jpldd.py:415]: use logger.error()

File: gluefactory/datasets/oxford_paris_mini_1view_jpldd.py [hunk 3]
```diff
@@ -90,6 +109,38 @@ def _init(self, conf):
...
 
+        augmentation_map = {
+           "dark": augmentations.DarkAugmentation,
+            "lg": augmentations.LGAugmentation,
+            "identity": augmentations.IdentityAugmentation
+        }
+
+        self.augmentation = augmentation_map[self.conf.load_features.augment.type]()
+
+    def download_binary_df_gt(self):
```
Comment (RaphaelKreft) [inline gluefactory/datasets/oxford_paris_mini_1view_jpldd.py:120]: I would prefer if we have everything in one download. It makes things easier. 
I can give you the directions to our current gt that contains everything so you can add the line gt to it and we can reupload. What do you think? (I would also be okay with keeping it that way if it works fine)

File: gluefactory/datasets/oxford_paris_mini_1view_jpldd.py [hunk 4]
```diff
@@ -60,8 +66,13 @@ class OxfordParisMiniOneViewJPLDD(BaseDataset):
             },
             "line_gt": {
                 "data_keys": ["deeplsd_distance_field", "deeplsd_angle_field"],
+                "use_binary_gt": False,
+                "thickness": 1,  # binary df line thickness
```
Comment (RaphaelKreft) [inline gluefactory/datasets/oxford_paris_mini_1view_jpldd.py:70]: give this a more expressive name ie. line_heatmap_thickness

File: gluefactory/datasets/oxford_paris_mini_1view_jpldd.py
Comment (RaphaelKreft) [inline gluefactory/datasets/oxford_paris_mini_1view_jpldd.py:1]: Wer could think about adding options like use_df, use_df as well so we can dynamically choose what to use. This becomes relevant when we ditch the df.

**Review by RaphaelKreft — COMMENTED** [review]
Please make the augmentations a SEPARATE PR as this has nothing to do with the binary line heatmap loading.
Also please addresss the questions and changes suggetsed.

File: gluefactory/datasets/oxford_paris_mini_1view_jpldd.py [hunk 5]
```diff
@@ -60,8 +66,13 @@ class OxfordParisMiniOneViewJPLDD(BaseDataset):
...
                 "enforce_threshold": 5.0,  # Enforce values in distance field to be no greater than this value
             },
+            "augment": {
```
Comment (RaphaelKreft) [inline gluefactory/datasets/oxford_paris_mini_1view_jpldd.py:73]: I would like to see that we can switch augmentation on and off. One of two possibilities:

1.  add a type "none" that will not apply any augmentations
2. add a k/v pair with key "do" and Boolean type as value. Then if its activated and which type is used are separately configurable

In both cases ADD DOCUMENTATION!! -> How is the augmentation applied(additional augmented images or probability of augmentation on each image load + what options for augmentation)

File: gluefactory/settings.py
Comment (ewgoedeke) [reply gluefactory/settings.py:1]: addressed: stopped tracking settings.py via gitignore (change will be effective in new PRs)

File: .vscode/launch.json
Comment (ewgoedeke) [reply .vscode/launch.json:1]: done. Change effective in replacement repos.

**Comment by RaphaelKreft** [comment]
This PR has been rreplaces by #38 and #37

**Pull request closed without merging**